app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,68 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+sealed class MovieCategoryLabelTextViewState();
+
+data class MovieCategoryTextString(val category: String) : MovieCategoryLabelTextViewState()
+data class MovieCategoryTextStringResource(val category: Int) : MovieCategoryLabelTextViewState()
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState,
+)
+
+@Composable
+fun MovieCategoryLabel(
+    item: MovieCategoryLabelViewState,
+    modifier: Modifier = Modifier,
+) {
+
+    val rememberState = remember { mutableStateOf(item.isSelected) }
+
+    Column(
+        modifier = modifier
+            .height(30.dp)
+    ) {
+        Text(
+            text = when (item.categoryText) {
+                is MovieCategoryTextString -> item.categoryText.category;
+                is MovieCategoryTextStringResource -> stringResource(id = item.categoryText.category)
+            },
+            fontSize = 16.sp,
+            textDecoration = if (rememberState.value) TextDecoration.Underline else TextDecoration.None,
```

Comment:
This works fine, but try doing it without textDecoration. Try to use a Box with a background and predefined height.

Reply:
Would the line then be in the background image?

Reply:
Not sure what do you mean by image, because there is no image in this component. 
The line would be an independent component (Box composable). Column is a layout which takes multiple composable functions and place them vertically. This is exactly what you need. Text will be the first composable, then you can add some spacing and then add Box with background and adjust size to be the wanted size of the underline.